python/pyspark/sql/dataframe.py

```diff
@@ -1648,8 +1650,30 @@ def toPandas(self):
         0    2  Alice
         1    5    Bob
         """
-        import pandas as pd
-        return pd.DataFrame.from_records(self.collect(), columns=self.columns)
+        if self.sql_ctx.getConf("spark.sql.execution.arrow.enable", "false").lower() == "true":
+            try:
+                import pyarrow
+                tables = self._collectAsArrow()
+                table = pyarrow.concat_tables(tables)
```

Comment:
Good point @leifwalsh , I have a test for an empty DataFrame in Java and a test for empty partition in Python, but there should one for this case also and it should definitely handle the error.  I'll put that in the next update.